contracts/cw-ics721-bridge/src/state.rs

```diff
@@ -16,17 +16,19 @@ pub const PO: PauseOrchestrator = PauseOrchestrator::new("c", "d");
 /// contract we have instantiated for that classID.
 pub const CLASS_ID_TO_NFT_CONTRACT: Map<ClassId, Addr> = Map::new("e");
 /// Maps cw721 contracts to the classID they were instantiated for.
-pub const NFT_CONTRACT_TO_CLASS_ID: Map<Addr, ClassId> = Map::new("f");
+pub const NFT_CONTRACT_TO_CLASS: Map<Addr, Class> = Map::new("f");
```

Comment:
could you say a little about why you chose to store the `Class` here instead of keeping it `ClassId` and requiring a second lookup in `CLASS_ID_TO_CLASS`. i feel a little nervous having two sources of truth for the `Class` data.

Reply:
Honestly, I don't have a good reason except my logic and because of how its set up:

1. When you ask for the classId from nft contract address, chances are you also want the class uri and data, we already have a struct ready for this, so why not return the whole struct instead of just class_id? it will reduce a lot of queries.
Thats the reason why I think we should save the whole class struct, instead of just the id.
2. if you already have a class id, you might want to query just class uri and data, without `CLASS_ID_TO_CLASS` storage, you would need to query for the nft contract first, and then get the class struct?
This is why I think we need `CLASS_ID_TO_CLASS` storage.

So basically combing those 2 reasons lead to having 2 source of truth for class.

BUT looking in the code this might be wrong, because on create_voucher, we might not have the nft contract address yet:
1. so I either change `NFT_CONTRACT_TO_CLASS` back to only hold the class_id, and keep `CLASS_ID_TO_CLASS` as 1 source of truth for class uri and data (but will always require 2 queries to get class uri and data)
2. or I move both saves into the mint callback, and pass the class along to it, which is really not the place for class stuff.

Seems like option 1 is better, thoughts?

Reply:
i think (1) is a good idea. nice catch